test/functional/test-amp-context.js

```diff
@@ -61,6 +59,43 @@ describe('3p ampcontext.js', () => {
     windowMessageHandler = undefined;
   });
 
+  it('should send message when report3pError_()', () => {
+    win.name = generateSerializedAttributes();
+    const context = new AmpContext(win);
+    expect(context).to.be.ok;
+
+    // Resetting since a message is sent on construction.
+    windowPostMessageSpy.reset();
+
+    const e = new Error();
+    e.message = 'message';
+
+    const messagePayload = {
+      sentinel: '1-291921',
+      type: MessageType.USER_ERROR,
+    };
+
+    const messageData = 'amp-' + JSON.stringify(messagePayload);
+    const message = {
+      source: context.client_.hostWindow_,
+      data: messageData,
+      message: e.message,
+    };
+    windowMessageHandler(message);
+
+    window.onerror = function(message, source, lineno, colno, error) {
```

Comment:
I see we use `describe` here, so please do not rewrite window function. This can potentially break other unrelated tests that depend on `window.onerror`. 
You can mock it in `win`, like what we did for `addEventListener`. 
Also this test here simply fake everything that we want to test. Can we use the `try/catch` we have before?

Reply:
Because for testing onerror, it will need to manually call the handler instead of how we dealt with addEventListener(). cc @cramforce

Reply:
I see. I am however still opposing overwriting `window.onerror` here, it would super hard to debug later. We can 1) rewrite the test to use `describes` or 2) make the onerror function a method and test that method instead.